### PR TITLE
Lifecycle toggle

### LINKED ISSUE #3647: Lifecycle toggle
## Is your feature request related to a problem?

*Please describe.*
I would like to be able to toggle different parts of lifecycle on and off to be able to clearly compare the numbers between dates

## Describe the solution you'd like
- add a toggle on the insights page when lifecycle is viewed

## Describe alternatives you've considered
leave it as is

### PR DESCRIPTION
**Pull request by liyiy**
## Changes

Addresses #3647 

Introduce lifecycle toggles that allow user to quickly compare "selected" lifecycles. By default all lifecycles are toggled on.

https://user-images.githubusercontent.com/25164963/114743017-465ad300-9d1a-11eb-927f-21ad93313f9a.mov

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests
- [ ] Migrations are safe to run at scale (e.g. PostHog Cloud) – present proof if not obvious

### CODE REVIEW

File: frontend/src/scenes/trends/trendsLogic.ts [hunk 1]
```diff
@@ -438,10 +455,20 @@ export const trendsLogic = kea<
...
-                actions.setVisibilityById({ [`${index}`]: true })
-                return { ...element, id: index }
-            })
+            let indexedResults
+            if (values.filters.insight !== ViewType.LIFECYCLE) {
+                indexedResults = values.results.map((element, index) => {
+                    actions.setVisibilityById({ [`${index}`]: true })
+                    return { ...element, id: index }
+                })
+            } else {
```
Comment (EDsCODE) [inline frontend/src/scenes/trends/trendsLogic.ts:464]: Good catch with this condition on refresh

**Review by EDsCODE — APPROVED** [review]
Good work! Looks like you need to fix prettier issues otherwise lgtm

**Comment by liyiy** [comment]
> Good work! Looks like you need to fix prettier issues otherwise lgtm

I ran `yarn prettier:check` locally and don't see errors indicating linting issues with the files I've affected, am I missing something?

**Comment by liyiy** [comment]
> > Good work! Looks like you need to fix prettier issues otherwise lgtm
> 
> I ran `yarn prettier:check` locally and don't see errors indicating linting issues with the files I've affected, am I missing something?

of course right after I post this I figure it out x)

**Pull request merged**